sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -127,20 +156,112 @@ class SingleStatementExec(
     origin.sqlText.get.substring(origin.startIndex.get, origin.stopIndex.get + 1)
   }
 
-  override def reset(): Unit = isExecuted = false
+  override def reset(): Unit = {
+    raisedError = false
+    errorState = None
+    error = None
+    rethrow = None
+    result = None // Should we do this?
+  }
+
+  def execute(session: SparkSession): Unit = {
```

Comment:
`override def` - though you'll catch it after you rebase to the new PR.

Reply:
Done.